### PR TITLE
CORE-5744 Membership API clean up

### PR DESCRIPTION
**Pull request by charlieR3**
Remove unused APIs. Tidied up APIs. Tidied up Kdocs. Removed unused kafka topics.

Runtime OS PR: https://github.com/corda/corda-runtime-os/pull/2017

### CODE REVIEW

**Review by nikinagy — DISMISSED** [review]
LGTM

**Review by nikinagy — DISMISSED** [review]
LGTM

**Review by yift-r3 — DISMISSED** [review]
LGTM

**Comment by charlieR3** [comment]
This conversation was highlighted to me before merging this: https://r3holdco.slack.com/archives/C039BDESELW/p1662046503131989

Added the change to this PR and rebased after API version change in runtime-os repo.

**Review by yift-r3 — DISMISSED** [review]
LGTM

**Comment by charlieR3** [comment]
Pending review from @corda/corda5-team-leads before being able to merge.

File: membership/src/main/kotlin/net/corda/v5/membership/MemberInfo.kt
Comment (alexey-kadyrov-r3) [inline membership/src/main/kotlin/net/corda/v5/membership/MemberInfo.kt]: Is it good to describe like that? As far as I remember all keys are "active" - can be used to sign. Would it be better to describe it as the latest? or To describe that the order of the keys would be with the latest generated at the top?

Comment (charlieR3) [reply membership/src/main/kotlin/net/corda/v5/membership/MemberInfo.kt]: Fair point. I'll change wording to `latest`

Comment (charlieR3) [reply membership/src/main/kotlin/net/corda/v5/membership/MemberInfo.kt]: Updated

**Review by alexey-kadyrov-r3 — DISMISSED** [review]
LGTM

File: ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/LedgerGroupParameters.kt
Comment (mnesbit) [inline ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/LedgerGroupParameters.kt]: Why is this not plural? We need to allow multiple notaries in the network

Comment (mnesbit) [inline ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/LedgerGroupParameters.kt]: I would expect key to still be plural form as we allow a list of notaries in the network

**Review by mnesbit — DISMISSED** [review]
One minor comment

File: ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/LedgerGroupParameters.kt
Comment (charlieR3) [reply ledger-obsolete/src/main/kotlin/net/corda/v5/ledger/obsolete/LedgerGroupParameters.kt]: Reverted. Originally changed to match design doc but plural would be more in line with other properties.

**Review by yift-r3 — APPROVED** [review]

**Review by alexey-kadyrov-r3 — APPROVED** [review]
Lgtm

**Pull request merged**
